subsys/bootloader/image/CMakeLists.txt

```diff
@@ -4,59 +4,50 @@
 # SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
 #
 
+if (NOT IMAGE)
+  if (CONFIG_SECURE_BOOT)
+    if (CONFIG_BOOTLOADER_MCUBOOT)
+      set(mcuboot_CONFIG_FW_METADATA "y" CACHE STRING "Build with FW_METADATA for B0")
+      set(mcuboot_CONFIG_BOOTED_BY_SECURE_BOOT "y" CACHE STRING "Booted by B0")
+      set(s0_addr $<TARGET_PROPERTY:partition_manager,PM_S0_ADDRESS>)
+      set(s1_addr $<TARGET_PROPERTY:partition_manager,PM_S1_ADDRESS>)
+    else()
+      if (CONFIG_SPM)
+        set(spm_CONFIG_FW_METADATA "y" CACHE STRING "Build with FW_METADATA for B0")
+        set(spm_CONFIG_BOOTED_BY_SECURE_BOOT "y" CACHE STRING "Booted by B0")
+        set(s0_addr $<TARGET_PROPERTY:partition_manager,PM_SPM_ADDRESS>)
+        set(s1_addr $<TARGET_PROPERTY:partition_manager,PM_SPM_ADDRESS>)
```

Comment:
make script handle empty s1_addr so that this line *and the one for app) is not needed